### PR TITLE
Added groupwiseconv and modified depthwise conv for common interface

### PR DESCRIPTION
**Pull request by arhik**
I am trying to add `groupcount` as a member for drafted changes here ->
https://github.com/FluxML/NNlib.jl/pull/146#issue-347196058

In short I am following @staticfloat 's suggestion here https://github.com/JuliaGPU/CuArrays.jl/pull/388#issuecomment-521697345

### CODE REVIEW

File: src/layers/conv.jl [hunk 1]
```diff
@@ -51,6 +51,7 @@ function (c::Conv)(x::AbstractArray)
   # TODO: breaks gpu broadcast :(
   # ndims(x) == ndims(c.weight)-1 && return squeezebatch(c(reshape(x, size(x)..., 1)))
   σ, b = c.σ, reshape(c.bias, map(_->1, c.stride)..., :, 1)
+  @show b
```
Comment (arhik) [inline src/layers/conv.jl]: Ignore this. I will remove it.

**Comment by dhairyagandhi96** [comment]
Looking forward to this! Thanks

**Comment by ToucheSir** [comment]
Following the breadcrumbs so this issue stays updated, per https://github.com/JuliaGPU/CuArrays.jl/pull/523#issuecomment-775758422 NNlib needs updating before we can revisit a layer in Flux.

**Comment by ToucheSir** [comment]
As of https://github.com/FluxML/Flux.jl/pull/1531, grouped convolutions are now possible with `Conv(...; groups=...)`.

**Pull request closed without merging**